rust/telemetry/src/feature_flags.rs

```diff
@@ -0,0 +1,144 @@
+use std::{sync::LazyLock, time::Duration};
+
+use anyhow::{bail, Context as _, Result};
+use parking_lot::RwLock;
+use serde::{Deserialize, Serialize};
+use tokio::runtime::Runtime;
+
+const POSTHOG_API_KEY: &str = "phc_uXXl56plyvIBHj81WwXBLtdPElIRbm7keRTdUCmk8ll";
+const RE_EVAL_DURATION: Duration = Duration::from_secs(5 * 60);
+
+static RUNTIME: LazyLock<Runtime> = LazyLock::new(init_runtime);
+
+// Process-wide storage of enabled feature flags.
+//
+// Defaults to everything off.
+static FEATURE_FLAGS: LazyLock<RwLock<FeatureFlags>> = LazyLock::new(RwLock::default);
+
+pub fn icmp_unreachable_instead_of_nat64() -> bool {
+    FEATURE_FLAGS.read().icmp_unreachable_instead_of_nat64
+}
+
+pub fn drop_llmnr_nxdomain_responses() -> bool {
+    FEATURE_FLAGS.read().drop_llmnr_nxdomain_responses
+}
+
+pub(crate) fn reevaluate(user_id: String) {
+    RUNTIME.spawn(async move {
+        let flags = decide(user_id)
+            .await
+            .inspect_err(|e| tracing::debug!("Failed to evaluate feature flags: {e:#}"))
+            .unwrap_or_default();
+
+        tracing::debug!(?flags, "Evaluated feature-flags");
+
+        *FEATURE_FLAGS.write() = flags;
+
+        sentry::Hub::main().configure_scope(|scope| {
+            scope.set_context("flags", sentry_flag_context(flags));
+        });
+    });
+}
+
+/// Initialize the runtime to use for evaluating feature flags.
+fn init_runtime() -> Runtime {
```

Comment:
I wonder if at some point it makes sense to have one thread that handles all non-IO-critical tasks and then the IO threads that need to remain performant. Not an issue now, but with each thread eating up 2-3 MB of stack I suppose it could be later on.

Reply:
We should tune the stack-size first I think.